earthfile2llb/converter.go

```diff
@@ -1007,10 +1007,10 @@ func (c *Converter) SaveImage(ctx context.Context, imageNames []string, pushImag
 				shouldPush := pushImages && si.DockerTag != "" && c.opt.DoSaves
 				shouldExportLocally := si.DockerTag != "" && c.opt.DoSaves
 				c.waitBlock().addSaveImage(si, c, shouldPush, shouldExportLocally)
-			} else {
-				c.mts.Final.SaveImages = append(c.mts.Final.SaveImages, si)
-			}
 
+				si.SkipBuilder = true
```

Comment:
This makes sense: if the wait block handles the export, the builder shouldn't. Also, this will make it easier to wrap the whole execution in an implied wait block when we get there.